drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -0,0 +1,705 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_IEEE802154_DRIVER_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ieee802154_cc13xx_cc26xx_subg);
+
+#include <device.h>
+#include <errno.h>
+#include <sys/byteorder.h>
+#include <net/ieee802154_radio.h>
+#include <net/ieee802154.h>
+#include <net/net_pkt.h>
+#include <random/rand32.h>
+#include <string.h>
+#include <sys/sys_io.h>
+
+#include <driverlib/rf_mailbox.h>
+#include <driverlib/rf_prop_mailbox.h>
+#include <driverlib/rfc.h>
+#include <inc/hw_ccfg.h>
+#include <inc/hw_fcfg1.h>
+#include <rf_patches/rf_patch_cpe_multi_protocol.h>
+
+#include <ti/drivers/rf/RF.h>
+
+#include "ieee802154_cc13xx_cc26xx_subg.h"
+
+static void ieee802154_cc13xx_cc26xx_subg_rx_done(
+	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data);
+static void ieee802154_cc13xx_cc26xx_subg_data_init(
+	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data);
+static int ieee802154_cc13xx_cc26xx_subg_stop(
+	const struct device *dev);
+static int ieee802154_cc13xx_cc26xx_subg_rx(
+	const struct device *dev);
+static void ieee802154_cc13xx_cc26xx_subg_setup_rx_buffers(
+	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data);
+
+DEVICE_DECLARE(ieee802154_cc13xx_cc26xx_subg);
+
+/* Overrides from SmartRF Studio 7 2.18.0 */
+static uint32_t overrides_sub_ghz[] = {
+	(uint32_t)0x00F788D3,
+	(uint32_t)0x4001405D,
+	(uint32_t)0x08141131,
+	ADI_2HALFREG_OVERRIDE(0, 16, 0x8, 0x8, 17, 0x1, 0x1),
+	HW_REG_OVERRIDE(0x6028, 0x001A),
+	HW_REG_OVERRIDE(0x609C, 0x0016),
+	(uint32_t)0x000188A3,
+	ADI_HALFREG_OVERRIDE(0, 61, 0xF, 0x8),
+	ADI_REG_OVERRIDE(0, 12, 0xF8),
+	(uint32_t)0xFFFFFFFF
+};
+
+/* Sub GHz power table */
+static const RF_TxPowerTable_Entry txPowerTable_sub_ghz[] = {
+	{ -20, RF_TxPowerTable_DEFAULT_PA_ENTRY(0, 3, 0, 2) },
+	{ -15, RF_TxPowerTable_DEFAULT_PA_ENTRY(1, 3, 0, 3) },
+	{ -10, RF_TxPowerTable_DEFAULT_PA_ENTRY(2, 3, 0, 5) },
+	{ -5, RF_TxPowerTable_DEFAULT_PA_ENTRY(4, 3, 0, 5) },
+	{ 0, RF_TxPowerTable_DEFAULT_PA_ENTRY(8, 3, 0, 8) },
+	{ 1, RF_TxPowerTable_DEFAULT_PA_ENTRY(9, 3, 0, 9) },
+	{ 2, RF_TxPowerTable_DEFAULT_PA_ENTRY(10, 3, 0, 9) },
+	{ 3, RF_TxPowerTable_DEFAULT_PA_ENTRY(11, 3, 0, 10) },
+	{ 4, RF_TxPowerTable_DEFAULT_PA_ENTRY(13, 3, 0, 11) },
+	{ 5, RF_TxPowerTable_DEFAULT_PA_ENTRY(14, 3, 0, 14) },
+	{ 6, RF_TxPowerTable_DEFAULT_PA_ENTRY(17, 3, 0, 16) },
+	{ 7, RF_TxPowerTable_DEFAULT_PA_ENTRY(20, 3, 0, 19) },
+	{ 8, RF_TxPowerTable_DEFAULT_PA_ENTRY(24, 3, 0, 22) },
+	{ 9, RF_TxPowerTable_DEFAULT_PA_ENTRY(28, 3, 0, 31) },
+	{ 10, RF_TxPowerTable_DEFAULT_PA_ENTRY(18, 2, 0, 31) },
+	{ 11, RF_TxPowerTable_DEFAULT_PA_ENTRY(26, 2, 0, 51) },
+	{ 12, RF_TxPowerTable_DEFAULT_PA_ENTRY(16, 0, 0, 82) },
+	{ 13, RF_TxPowerTable_DEFAULT_PA_ENTRY(36, 0, 0, 89) },
+	{ 14, RF_TxPowerTable_DEFAULT_PA_ENTRY(63, 0, 1, 0) },
+	RF_TxPowerTable_TERMINATION_ENTRY
+};
+
+static inline int ieee802154_cc13xx_cc26xx_subg_channel_to_frequency(uint16_t channel, uint16_t *frequency, uint16_t *fractFreq)
```

Comment:
don't use camel case for your own code (one coming from the HAL is obviously fine).

Btw, you seem to follow the 80 chars limits properly almost everywhere but it in some places like this one. Please keep consistency all over the code.